src/index.js

```diff
@@ -0,0 +1,118 @@
+import './style.css';
+import updateTaskStatus from './task_status_update.js';
+
+// getting elements
+const form = document.getElementById('add-to-list');
+const listItems = document.getElementById('to-do-list');
+const btnClear = document.querySelector('.btn-clear');
+
+// array for storing objects of to-do tasks
+const taskArr = [];
+
+// function for display added task to list
+const displayTask = (task) => {
+  const listItem = `
+  <li>
+    <div class="check">
+      <input type="checkbox" name="checkbox" class="checkbox" id="${task.description}">
+      <input type="text" class="task-description" name="${task.description}" class="task-name" id="task-name" value="${task.description}">
+    </div>
+    <div class="actions">
+      <i class="fa-solid fa-pen-to-square edit"></i>
+      <i class="fa-solid fa-trash-can del"></i>
+    </div>
+  </li>`;
+  return listItem;
+};
+
+window.addEventListener('DOMContentLoaded', () => {
+  const tasks = JSON.parse(localStorage.getItem('tasks'));
+  // displaying tasks on window loading
+  tasks.forEach((task) => {
+    const listItem = displayTask(task);
+    listItems.insertAdjacentHTML('beforeend', listItem);
+  });
+  // setting checkbox value to checked on window refresh if status is completed
+  const completedTasksIndex = tasks.filter((task) => task.completed === true);
+  for (let i = 0; i < completedTasksIndex.length; i += 1) {
+    for (let j = 0; j < (listItems.children).length; j += 1) {
+      if (j === (completedTasksIndex[i].index - 1)) {
+        listItems.children[j].children[0].children[0].checked = true;
+      }
+    }
+  }
+});
+
+// function for adding task to list
+const addTask = (task) => {
+  const taskObj = {};
+  taskObj.index = taskArr.length + 1;
+  taskObj.description = task;
+  taskObj.completed = false;
+  const listItem = displayTask(taskObj);
+  listItems.insertAdjacentHTML('beforeend', listItem);
+  taskArr.push(taskObj);
+  localStorage.setItem('tasks', JSON.stringify(taskArr));
+};
+
+// deleting task
+const deleteTask = (task, element) => {
+  const taskName = task.children[0].children[1].value;
+  const tasks = JSON.parse(localStorage.getItem('tasks'));
+  const taskIndex = tasks.findIndex((x) => x.description === taskName);
+  tasks.splice(taskIndex, 1);
+  tasks.forEach((item, ind) => {
+    item.index = ind + 1;
+  });
+  localStorage.setItem('tasks', JSON.stringify(tasks));
```

Comment:
Maybe you can create some functions to interact with localStorage? That for don't repeat the code to set an get the data from the localStorage.

![image](https://user-images.githubusercontent.com/53025062/200393403-7fb06898-b6b4-4826-bf84-a475c9d89379.png)